### PR TITLE
Elasticsearch sink refactor

### LINKED ISSUE #998: Refactor Elasticsearch sink
Elasticsearcy type mapping highly depends on kubernetes event.Event struct. As part of kubernetes plugin extraction, now contracted struct is api.Message and we need to refactor this mapping.

### PR DESCRIPTION
**Pull request by huseyinbabal**
## Description

Changes proposed in this pull request:

- Elasticsearch sink is updated to accept new contract `api.Message`
- Introduced rawData param to send original data to sinks
- Collector is updated to enable plugins for also sinks

### Build plugins
```bash
PLUGIN_DOWNLOAD_URL_BASE_PATH=http://localhost:8080/plugin-dist make gen-plugins-index
```

### Run plugin server
```bash
npx serve --listen 8080
```

### Run Botkube with the following  config

```bash
export BOTKUBE_CONFIG_PATHS=/tmp/botkube.yml
export BOTKUBE_SETTINGS_KUBECONFIG=${USER}/.kube/config

go run cmd/botkube/main.go
```

```yaml
## /tmp/botkube.yml
communications:
  'default-group':
    elasticsearch:
      enabled: true
      server: 'elastic_url'
      username: user
      password: pass
      indices:
        'default':
          name: botkube
          type: kubernetes
          shards: 1
          replicas: 1
          bindings:
            sources: 
              - k8s-events
              - k8s-updates


settings:
  clusterName: gke-playground
  configWatcher: true
  log:
    level: debug
  lifecycleServer:
    deployment:
      name: botkube
      namespace: botkube
    port: "2113"

configWatcher:
  enabled: false
  initialSyncTimeout: 0

sources:
  'k8s-events':
    displayName: "K8s recommendations"
    'botkube/kubernetes':
      enabled: true
      config:
        log:
          level: debug
        recommendations:
          pod:
            noLatestImageTag: true
            labelsSet: true
          ingress:
            backendServiceValid: false
            tlsSecretValid: false
        namespaces:
          include:
            - botkube
        event:
          types:
            - create
            - update

  'k8s-updates':
    displayName: "K8s ConfigMaps updates"
    'botkube/kubernetes':
      enabled: true
      config:
        log:
          level: debug
        namespaces:
          include:
            - default
        event:
          types:
            - create
            - update
            - delete
        resources:
          - type: v1/configmaps
            namespaces:
              include:
                - botkube
            event: # overrides top level `event` entry
              types:
                - update
            updateSetting:
              includeDiff: true
              fields:
                - data

plugins:
  repositories:
    botkube:
      url: http://localhost:8080/plugins-index.yaml

```

### Create a pod to see a notification
```bash
kubectl run nginx_es --image=nginx:latest -n botkube
```

## Related issue(s)
Fixes #998

### CODE REVIEW

File: pkg/sink/elasticsearch.go [hunk 1]
```diff
@@ -174,38 +172,29 @@ func (e *Elasticsearch) flushIndex(ctx context.Context, indexCfg config.ELSIndex
...
+	e.log.Debugf(">> Sending to Elasticsearch: %+v", rawData)
 
 	errs := multierror.New()
 	for _, indexCfg := range e.indices {
-		if !sliceutil.Intersect(indexCfg.Bindings.Sources, eventSources) {
-			continue
-		}
-
-		err := e.flushIndex(ctx, indexCfg, event)
+		err := e.flushIndex(ctx, indexCfg, rawData)
```
Comment (pkosiec) [inline pkg/sink/elasticsearch.go:186]: That's might be problematic as I know we needed to ignore some properties from the `event.Event` as ELS didn't accept some of the nested struct data. Just saying, but it should work OK as long as all events have that in mind 👍

**Review by pkosiec — COMMENTED** [review]
Please see the review of https://github.com/kubeshop/botkube/pull/1008. Thanks!

File: pkg/sink/elasticsearch.go (hunk 1 above)
Comment (huseyinbabal) [reply pkg/sink/elasticsearch.go:186]: I tested this for kubernetes plugin and the data was shipped to elasticsearch, but I think we need to state this in the docs maybe, "we don't have control on data structure, so if you have es integration, you might get error due to your complex nested data structure coming from custom plugin".

File: pkg/sink/elasticsearch.go [hunk 2]
```diff
@@ -196,8 +196,21 @@ func (e *Elasticsearch) SendMessage(ctx context.Context, event event.Event, even
 }
 
 // SendEvent is no-op.
-func (e *Elasticsearch) SendEvent(_ context.Context, _ any, _ []string) error {
-	return nil
+func (e *Elasticsearch) SendEvent(ctx context.Context, rawData any, sources []string) error {
+	e.log.Debugf(">> Sending to Elasticsearch: %+v", rawData)
+
+	errs := multierror.New()
+	for _, indexCfg := range e.indices {
```
Comment (pkosiec) [inline pkg/sink/elasticsearch.go:203]: This logic is invalid - we send events to all indexes and `sources` are not used. We should do

```go
		if !sliceutil.Intersect(indexCfg.Bindings.Sources, eventSources) {
			continue
		}
```
instead.

Basically please use logic from `SendMessage` in `SendEvent`, and remove `SendMessage`. Thanks!

**Review by pkosiec — APPROVED** [review]
LGTM but please apply comments before merge. Thanks!

File: pkg/sink/elasticsearch.go (hunk 2 above)
Comment (huseyinbabal) [reply pkg/sink/elasticsearch.go:203]: Actually, I removed SendMessage now, and each es flush index is for one source, so I removed source part

**Pull request merged**